src/ferc_xbrl_extractor/instance.py

```diff
@@ -253,3 +200,125 @@ def from_xml(cls, elem: Element) -> "Fact":
             c_id=elem.attrib["contextRef"],
             value=elem.text,
         )
+
+
+class Instance:
+    """
+    Class to encapsulate a parsed instance.
+
+    This class should be constructed using the InstanceBuilder class. Instance wraps
+    the contexts and facts parsed by the InstanceBuilder, and is used to construct
+    dataframes from fact tables.
+    """
+
+    def __init__(
+        self,
+        contexts: Dict[str, Context],
+        facts: Dict[str, List[Fact]],
+        filing_name: str,
+    ):
+        """
+        Construct Instance from parsed contexts and facts.
+
+        This will use a dictionary to map contexts and facts to the Axes defined for
+        the relevant contexts. This makes it easy to identify which facts might belong
+        in a specific fact table.
+
+        Args:
+            contexts: Dictionary mapping context ID to contexts.
+            facts: Dictionary mapping context ID to facts.
+            filing_name: Name of parsed filing.
+        """
+        # This is a nested dictionary of dictionaries to locate facts by context
+        self.instant_facts = {}
+        self.duration_facts = {}
+
+        self.filing_name = filing_name
+
+        for c_id, context in contexts.items():
+            axes = tuple([dim.name for dim in context.entity.dimensions])
+            if context.period.instant:
+                if axes not in self.instant_facts:
+                    self.instant_facts[axes] = {}
+
+                self.instant_facts[axes][context] = facts[c_id]
+            else:
+                if axes not in self.duration_facts:
+                    self.duration_facts[axes] = {}
+
+                self.duration_facts[axes][context] = facts[c_id]
+
+    def get_facts(self, instant: bool, axes: List[str]) -> Dict[Context, List[Fact]]:
+        """
+        Return a dictionary that maps Contexts to a list of facts for each context.
+
+        Args:
+            instant: Get facts with instant or duration period.
+            axes: List of axis names defined for fact table.
+        """
+        if instant:
+            facts = self.instant_facts.get(tuple(axes), {})
+        else:
+            facts = self.duration_facts.get(tuple(axes), {})
+
+        return facts
+
+
+class InstanceBuilder:
```

Comment:
What is the benefit of adding an `InstanceBuilder` over keeping the parsing logic within `Instance`?

Reply:
Ah got it. Thanks!